dao/mongodb.rya/src/main/java/org/apache/rya/mongodb/iter/AggregationUtil.java

```diff
@@ -0,0 +1,322 @@
+     */
+    private AggregationUtil() {
+    }
+
+    /**
+     * Creates a MongoDB $redact aggregation pipeline that only include
+     * documents whose document visibility match the provided authorizations.
+     * All other documents are excluded.
+     * @param authorizations the {@link Authorization}s to include in the
+     * $redact. Only documents that match the authorizations will be returned.
+     * @return the {@link List} of {@link DBObject}s that represents the $redact
+     * aggregation pipeline.
+     */
+    public static List<DBObject> createRedactPipeline(final Authorizations authorizations) {
+        if (MongoDbRdfConstants.ALL_AUTHORIZATIONS.equals(authorizations)) {
+            return Lists.newArrayList();
+        }
+        final List<String> authAndList = authorizations.getAuthorizationsStrings();
+
+        // Generate all combinations of the authorization strings without repetition.
+        final List<List<String>> authOrList = createCombinations(authorizations.getAuthorizationsStrings());
+
+        final String documentVisibilityField = "$" + SimpleMongoDBStorageStrategy.DOCUMENT_VISIBILITY;
+
+        final BasicDBObject setIsSubset =
+            setIsSubsetNullSafe(
+                documentVisibilityField,
+                authAndList.toArray()
+            );
+
+        final BasicDBObject setIntersectionExists =
+            gt(
+                size(
+                    setIntersection(
+                        documentVisibilityField,
+                        authOrList.toArray()
+                    )
+                ),
+                0
+            );
+
+        final BasicDBObject orExpression = or(setIsSubset, setIntersectionExists);
+
+        final List<DBObject> pipeline = new ArrayList<>();
+        pipeline.add(
+            redact(
+               orExpression,
+               "$$DESCEND",
+               "$$PRUNE"
+            )
+        );
+
+        return pipeline;
+    }
+
+    /**
+     * Creates all combinations of the values that are of the size of value
+     * array or smaller without repetition.
+     * @param values the {@link List} of values to create combinations from.
+     * @return the {@link List} of combinations.
+     */
+    public static <T> List<List<T>> createCombinations(final List<T> values) {
+        final List<List<T>> allCombinations = new ArrayList<>();
+        for (int i = 1; i <= values.size(); i++) {
+            allCombinations.addAll(createCombinations(values, i));
+        }
+        return allCombinations;
+    }
+
+    /**
+     * Creates all combinations of the values that are of the specified size
+     * without repetition.
+     * @param values the {@link List} of values to create combinations from.
+     * @param size the size of the combinations.
+     * @return the {@link List} of combinations.
+     */
+    public static <T> List<List<T>> createCombinations(final List<T> values, final int size) {
```

Comment:
why not do an Auth -> bit, then do bit comparison?  This seems potentially expensive, there's gotta be a better way to do ordering comparisons